### PR TITLE
Adds promises to dvbjs

### PR DESCRIPTION
**Pull request by florianschmidt1994**
* All dvbjs functions should now successfully return a Promise.
Example:
```javascript
var data = dvb.pins(5654791, 4620310, 5657216, 4623119, 'stop') // Returns a chainable Promise
    .then(function(data) { /* Do something with data here */ })
    .catch(function(err) { /* Catch possible errors here */ });
```
* Thanks to bluebirds `Promise.nodeify(callback)` it is still possible to use dvbjs as always, using the nodejs callback `dvb.find(..., function(err, data) { /* Handle err and data here */ })` pattern.

* Adds `dvb.pins`, which resolves into a list of pins for a given bounding box and pintypes

* Adds `dvb.address`, which resolves into an object containing an address and a city for given coordinates

* Adds `dvb.coords`, which resolves into an array which contains the lat, lng coordinates for given ID of a POI

### CODE REVIEW

**Comment by rebugger** [comment]
Adding promises and keeping callbacks? Awesome!


I think we might need to think about API and versioning since the last pull requests had a lot of changes. 
Do you have any recommendations how to increase the version due to your changes? 3.2.1? 

In the develop branch i started splitting up the library in smaller files to keep better maintainability.
My thought was to keep development in the develop-branch and release to the master-branch, but I'm not too familiar with best practices on github yet.
I guess I have to merge your great changes in there.

File: test/test.js [hunk 1]
```diff
@@ -123,3 +123,69 @@ describe('dvb.find "Zellescher Weg"', function() {
         });
     });
 });
+
+describe('dvb.pins "5654791, 4620310, 5657216, 4623119, stop"', function() {
+
```
Comment (rebugger) [inline test/test.js]: You can store the responses to test/data and mock these using mockRequest(), this speeds up and stabilizes the unit-tests.

**Comment by florianschmidt1994** [comment]
I think 3.2.1 will do, as there are a bunch of improvements and additional features, but no major changes concerning the API itself.

Using something like develop for development and master for releases seems to be a feasible solution. I myself like the solution [git flow](https://de.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) provides, even though this could be kind of overkill here. 

There is one downside regarding my pull request. As i don't not how to stub the new request-promise library, the tests currently call the real API, making them slow again

File: lib/dvbjs.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 'use strict';
 
 var request = require('request');
```
Comment (rebugger) [inline lib/dvbjs.js]: This isn't needed anymore since you are using request-promise

File: test/test.js (hunk 1 above)
Comment (florianschmidt1994) [inline test/test.js]: I'd love too, but as of now I couldn't figure out how to stub the newly introduced request-promise library. Ideas are very welcome!

File: lib/dvbjs.js [hunk 3]
```diff
@@ -131,30 +155,130 @@ dvb.find = function(searchString, callback) {
...
+            'name' : dataAsString.split('||')[1].split('|')[0],
+            'coords' : [
+                dataAsString.split('||')[1].split('|')[1],
+                dataAsString.split('||')[1].split('|')[2]
+            ],
+            '?' : dataAsString.split('||')[1].split('|')[3]
+        };
+        return jsonElem;
+
+    } else if (_.contains(['poi', 'rentabike', 'ticketmachine', 'carsharing', 'parkandride'], pintypes)) {
```
Comment (rebugger) [inline lib/dvbjs.js]: How about replacing it with
```
-1 !== ['poi', 'rentabike', 'ticketmachine', 'carsharing', 'parkandride'].indexOf(pintypes)
```
and dropping the lodash dependency? Not very speaking but less dependencies.

File: test/test.js (hunk 1 above)
Comment (rebugger) [inline test/test.js]: Ok, i'll look into it and can add it later.

File: lib/dvbjs.js [hunk 4]
```diff
@@ -53,75 +53,99 @@ dvb.route = function(origin, destination, time, deparr, callback) {
...
+            name_destination       : destination,
+            nameState_destination  : 'empty',
+            itdTripDateTimeDepArr  : deparr,
+            itdTimeHour            : hour,
+            idtTimeMinute          : minute,
+            outputFormat           : 'JSON',
+            coordOutputFormat      : 'WGS84',
+            coordOutputFormatTail  : '0'
+        }    
+    };
```
Comment (kiliankoe) [inline lib/dvbjs.js]: ❤️

**Comment by kiliankoe** [comment]
This is too awesome 😳

File: lib/dvbjs.js (hunk 3 above)
Comment (florianschmidt1994) [inline lib/dvbjs.js]: Seems to me that lodash could be something worth keeping. For now it is only used in this line of code, but i remember from previous projects that i'm using lodash a lot.

File: lib/dvbjs.js (hunk 2 above)
Comment (florianschmidt1994) [inline lib/dvbjs.js]: Fixed in https://github.com/kiliankoe/dvbjs/commit/fb96b9480c3422afdfce7290da16e3630fbb419f

**Comment by kiliankoe** [comment]
Since I know Flo anyways and you guys are awesome, I just added him as a collaborator as well :)

**Pull request merged**